### ISSUE TITLE
Many li elements crashes react-dom

### ISSUE THREAD

**Issue by Stan64** [remarkjs/react-markdown#229]
Maximum call stack size exceeded

So we had a user type `* 1. * 1. * 1. * 1. * 1. * 1. * 1. * 1. * 1. * 1. * 1. * 1. * 1. * 1. * 1. * 1. * 1. * 1. * Inception`
Followed by another posts with a couple more lists and the website disappeared until I removed the data from the database.

```
react-dom.production.min.js:4200 RangeError: Maximum call stack size exceeded
    at get-definitions.js:5
    at Array.reduce (<anonymous>)
    at e (get-definitions.js:5)
    at get-definitions.js:13
    at Array.reduce (<anonymous>)
    at e (get-definitions.js:5)
    at get-definitions.js:13
    at Array.reduce (<anonymous>)
    at e (get-definitions.js:5)
    at get-definitions.js:13
Io @ react-dom.production.min.js:4200
```

Works fine in Firefox but crashes in Chrome.

**Comment by rexxars** [comment]
Which version are you using?

**Comment by Stan64** [comment]
4.0.2

**Comment by rexxars** [comment]
Any chance you have the original source? This has to do with definitions (`[1]: Definition`), but I can't seem to reproduce it

**Comment by Stan64** [comment]
I can't reproduce it either, I asked the same user to try again, I'll tell you if he answers.

I'm using a plugin `remark-breaks` if that has anything to do with it.

```
<ReactMarkdown source={body} plugins={[ breaks ]} />
```

**Comment by Stan64** [comment]
@rexxars Were you able to reproduce it?

**Comment by rexxars** [comment]
Yes - I have not had time to look at it yet due to Real Life™️  - feel free to take a stab at it if you want

**Comment by ChristianMurphy** [comment]
#428 appears to resolve the initial case, https://github.com/remarkjs/remark/issues/439 should resolve the larger case with even more list items.

**Comment by wooorm** [comment]
I managed to double the allowed deeply nested nodes, so this one should be fixed. But, users can always upload longer pathological stuff that get your code to crash. You may want to have a fallback for when they do, such as display the plain text.

**Comment by staltz** [comment]
@wooorm I released `mdast-add-list-metadata@1.1.0`, thanks for the heads up.

**Comment by wooorm** [comment]
Awesome, thanks!! ✨✨
